torch/_inductor/scheduler.py

```diff
@@ -2145,6 +2193,21 @@ def add_user(
             for buf in node.get_outputs():
                 buf.set_users(name_to_users[buf.get_name()].items)
 
+        # compute and copy user information of donated buffer
+        for node in self.nodes:
+            for read in node.read_writes.reads:
+                if (
+                    read.name in V.graph.graph_inputs
+                    and isinstance(
+                        V.graph.graph_inputs[read.name].data.data, ir.DonatedBuffer  # type: ignore[attr-defined]
+                    )
+                    and not isinstance(read, WeakDep)
+                ):
+                    add_user(read.name, node, node.can_inplace(read))
```

Comment:
Why do we need this loop ? aren't all these reads already being added in https://github.com/pytorch/pytorch/blob/be51941aae41586429eb888debc3a28fc3961dc9/torch/_inductor/scheduler.py#L2141-L2145?